### PR TITLE
Add scripts and targets for manpages and yamldocs

### PR DESCRIPTION
**Pull request by tiborvass**
Signed-off-by: Tibor Vass <tibor@docker.com>

### CODE REVIEW

File: docs/yaml/generate.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
```
Comment (dnephin) [inline docs/yaml/generate.sh]: Could this be moved to `scripts/docs/generate-yaml.sh` to be consistent with our other scripts?

File: man/generate.sh [hunk 2]
```diff
@@ -7,6 +7,16 @@ set -eu
 
 mkdir -p ./man/man1
 
+MD2MAN_REPO=github.com/cpuguy83/go-md2man
```
Comment (dnephin) [inline man/generate.sh]: And this could be `scripts/docs/generate-man.sh` to be consistent with our other scripts.

File: Makefile [hunk 3]
```diff
@@ -3,20 +3,22 @@
 #
 all: binary
 
+PKGS := $(shell go list ./... | grep -vE '/vendor/|github.com/docker/cli/man|github.com/docker/cli/docs/yaml$$')
```
Comment (dnephin) [inline Makefile]: I don't think we need to exclude `man/` or `docs/` from unit testing anymore , now that we have all the dependencies installed in `vendor/`.

Comment (tiborvass) [inline Makefile]: @dnephin sure, I will simply duplicate it in this Makefile, but I do want to remove `grep -v vendor` in unit-test-with-coverage, and replace it with `$@`.

File: Makefile [hunk 4]
```diff
@@ -6,17 +6,17 @@ all: binary
...
 clean:
-	rm -rf ./build/* cli/winresources/rsrc_*
+	rm -rf ./build/* cli/winresources/rsrc_* ./man/man[1-9] docs/yaml/gen
 
 # run go test
 # the "-tags daemon" part is temporary
 .PHONY: test
 test:
-	./scripts/test/unit $(shell go list ./... | grep -v /vendor/)
+	./scripts/test/unit $(shell go list ./... | grep -vE '/vendor/|github.com/docker/cli/man|github.com/docker/cli/docs/yaml$$')
```
Comment (dnephin) [inline Makefile]: This no longer needs to ignore `docs/` or `man/`

**Pull request merged**